### PR TITLE
Feature/ci template lint

### PR DESCRIPTION
**Pull request by fabmiz**
- Ticket: []
- Feature flag: n/a

## Purpose

Enable template linting on travis.
Currently, template lint is only run in a pre-commit hook that can be bypassed with
`--no-verify`

## Summary of Changes
Addressed:
- [`required-button-type`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/require-button-type.md)
- [`no-negated-condition`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-negated-condition.md)
- [`quotes`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/quotes.md)
- [`no-quoteless-attributes`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-quoteless-attributes.md)
- [`no-implicit-this`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-implicit-this.md)
- [`no-bare-strings`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-bare-strings.md)
- [`no-inline-styles`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-inline-styles.md)
- [`link-rel-noopener`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/link-rel-noopener.md)
- [`no-duplicate-attributes`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-duplicate-attributes.md)
- [`table-groups`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/table-groups.md)
-  [`link-href-attributes`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/link-href-attributes.md)
- Remove unnecessary `template-lint-ignore` *

## QA Notes
N/A Dev tested.

### CODE REVIEW

File: lib/analytics-page/addon/application/styles.scss [hunk 1]
```diff
@@ -53,3 +53,9 @@
 .PrivateProject {
     padding: 1em 3em 2em;
 }
+
+:global(.btn.btn-link).button-link {
+    padding: 0;
+    margin: 0;
+    border: 0;
+}
```
Comment (adlius) [inline lib/analytics-page/addon/application/styles.scss:61]: Why is this needed?

File: lib/app-components/addon/components/project-contributors/list/item/template.hbs [hunk 2]
```diff
@@ -107,9 +108,10 @@
     <div class='hidden-xs col-sm-2 text-center'>
         <button
             data-test-project-contributors-list-item-remove-button
-            {{action @removeContributor @contributor}}
+            type='button'
             class='btn btn-danger btn-sm'
             disabled={{not this.canRemove}}
+            {{action @removeContributor @contributor}}
```
Comment (adlius) [inline lib/app-components/addon/components/project-contributors/list/item/template.hbs:114]: Can we use element modifiers here?

File: lib/app-components/addon/components/search-paginator/template.hbs [hunk 3]
```diff
@@ -3,9 +3,10 @@
         <li local-class='li'>
             <button
                 data-test-page={{item.text}}
+                type='button'
                 local-class={{if (eq this.current item.text) 'active'}}
-                {{action (or item.action 'setPage') item.text}}
                 disabled={{item.disabled}}
+                {{action (or item.action 'setPage') item.text}}
```
Comment (adlius) [inline lib/app-components/addon/components/search-paginator/template.hbs:9]: Is it that element modifier doesn't take arguments?

File: lib/app-components/addon/components/submit-section-buttons/template.hbs [hunk 4]
```diff
@@ -1,13 +1,24 @@
 {{#if this.showDiscard}}
-    <button data-test-submit-section-discard class='btn btn-default' {{action this.discard}}>
+    <button
+        data-test-submit-section-discard
+        type='button'
+        class='btn btn-default'
+        {{action this.discard}}
```
Comment (adlius) [inline lib/app-components/addon/components/submit-section-buttons/template.hbs:6]: Element modifier here?

File: lib/app-components/addon/components/submit-section-buttons/template.hbs [hunk 5]
```diff
@@ -1,13 +1,24 @@
...
         {{/if}}
     </button>
 {{/if}}
-<button data-test-submit-section-continue class='btn btn-primary' {{action this.continue}} disabled={{this.continueDisabled}}>
+<button
+    data-test-submit-section-continue
+    type='button'
+    class='btn btn-primary'
+    disabled={{this.continueDisabled}}
+    {{action this.continue}}
```
Comment (adlius) [inline lib/app-components/addon/components/submit-section-buttons/template.hbs:20]: Same here.

File: lib/app-components/addon/components/submit-section/complete/template.hbs [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
-        {{yield}}
-        <p
-            local-class='edit-message'
-            class='text-smaller'
-            hidden={{not this.editable}}
+    {{#if this.editable}}
+        <div
+            data-test-submit-section-click-to-edit
+            role='button'
+            {{action 'edit'}}
```
Comment (adlius) [inline lib/app-components/addon/components/submit-section/complete/template.hbs:6]: element modifier?

File: lib/collections/addon/components/collections-submission/template.hbs [hunk 7]
```diff
@@ -190,9 +192,10 @@
                     <button
                         data-test-collections-submit-submit-button
                         data-analytics-name='Submit'
+                        type='button'
                         class='btn btn-success'
-                        {{action this.setShowSubmitModal}}
                         disabled={{not (eq this.activeSection this.sections.submit)}}
+                        {{action this.setShowSubmitModal}}
```
Comment (adlius) [inline lib/collections/addon/components/collections-submission/template.hbs:198]: Same here.

File: lib/app-components/addon/components/project-contributors/list/item/template.hbs (hunk 2 above)
Comment (aaxelb) [reply lib/app-components/addon/components/project-contributors/list/item/template.hbs:114]: at risk of creeping scope, maybe should consider enabling [`no-action`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-action.md)?

File: translations/en-us.yml [hunk 8]
```diff
@@ -942,6 +942,7 @@ registries:
     discover:
         page_title: Search
         SHARE: SHARE
+        registrations: Registrations
```
Comment (adlius) [inline translations/en-us.yml:945]: Should this be in quotes?

File: translations/en-us.yml [hunk 9]
```diff
@@ -1214,7 +1215,7 @@ registries:
...
+        notice: "Please keep in mind that:\n                <ul>\n                    <li>\n                        Registrations cannot be modified or deleted once completed.\n                    </li>\n                    <li>\n                        The content and version history of Wiki and OSF Storage will be copied to the registration.\n                    </li>\n                    <li>\n                        This project contains links to other projects. These links will be copied into your registration, but the\n                        projects that they link to will not be registered. If you wish to register the linked projects, they must be\n                        registered separately. <a href='{learnMoreLink}' target='_blank' rel='noopener noreferrer'>Learn more about links</a>.\n                    </li>\n                </ul>"
```
Comment (adlius) [inline translations/en-us.yml:1218]: Single quote?

File: lib/collections/addon/components/discover-page/active-filters/filter/template.hbs [hunk 10]
```diff
@@ -8,6 +8,7 @@
         <span aria-label={{@ariaLabel}}>
             <button
                 data-test-active-filter-remove-button
+                type='button'
                 class='fa fa-times-circle'
                 local-class='removeActiveFilter'
                 {{action this.removeFilterItem}}
```
Comment (adlius) [inline lib/collections/addon/components/discover-page/active-filters/filter/template.hbs:14]: Could use element modifier here as well. But since you're not touching this line, probably not very important now.

File: lib/osf-components/addon/components/bs-alert/template.hbs [hunk 11]
```diff
@@ -1,6 +1,11 @@
 {{#unless this.hidden}}
     {{#if this.dismissible}}
-        <button class='close' aria-label={{t 'general.close'}} {{action 'dismiss'}}>
+        <button
+            type='button'
+            class='close'
+            aria-label={{t 'general.close'}}
+            {{action 'dismiss'}}
```
Comment (adlius) [inline lib/osf-components/addon/components/bs-alert/template.hbs:7]: Same here.

File: lib/osf-components/addon/components/files/selected-list/template.hbs [hunk 12]
```diff
@@ -7,7 +7,7 @@
             <AnimatedEach
                 @items={{@selectedFiles}}
                 @use={{this.transition}}
-                @duration=500
+                @duration='500'
```
Comment (adlius) [inline lib/osf-components/addon/components/files/selected-list/template.hbs:10]: Got a question: for these type of parameters that expect integers/numbers, can we use `{{500}}`? `'500'` makes it look like a string.

File: lib/osf-components/addon/components/new-project-modal/template.hbs [hunk 13]
```diff
@@ -42,31 +42,33 @@
...
+                        <tbody>
+                            <tr data-test-institution-button-row >
+                                {{#each this.institutions as | institution |}}
+                                    <td>
+                                        <OsfButton
+                                            data-test-institution-selected='{{if (if-filter institution this.selectedInstitutions) 'selected' 'not-selected'}}'
+                                            data-analytics-name='select_institution'
+                                            local-class='NewProject__institution-button'
+                                            @type='link'
+                                            @onClick={{action this.selectInstitution institution}}
```
Comment (adlius) [inline lib/osf-components/addon/components/new-project-modal/template.hbs:54]: Could use element modifier here.

File: lib/osf-components/addon/components/osf-navbar/x-links/hyper-link/template.hbs [hunk 14]
```diff
@@ -7,8 +7,8 @@
     ) as |ctx|}}
         <ctx.link
             data-analytics-name='{{@analyticsLabel}}'
-            data-test-hyper-link=@testSelectorVal
-            class=@class
+            data-test-hyper-link={{@testSelectorVal}}
```
Comment (adlius) [inline lib/osf-components/addon/components/osf-navbar/x-links/hyper-link/template.hbs:10]: Does that mean this never worked before?

File: lib/osf-components/addon/components/sign-up-form/template.hbs [hunk 15]
```diff
@@ -52,33 +51,29 @@
...
+            <ValidatedInput::Recaptcha
                 data-test-recaptcha
-                model=this.userRegistration
-                valuePath='recaptchaResponse'
-                shouldShowMessages=this.didValidate
-                bindReset=(action (mut this.resetRecaptcha))
-            }}
-            {{! template-lint-enable no-implicit-this }}
+                @model={{this.userRegistration}}
+                @valuePath='recaptchaResponse'
```
Comment (adlius) [inline lib/osf-components/addon/components/sign-up-form/template.hbs:73]: Can we always use `{{}}`? For example, can this be `{{'recaptchaResponse'}}`?

File: lib/registries/addon/components/registries-discover-search/x-sidebar/template.hbs [hunk 16]
```diff
@@ -1,18 +1,18 @@
...
-        <div class="col-xs-6">
+<div local-class='Sidebar__ActiveFilters' class='well well-sm'>
+    <div class='row'>
+        <div class='col-xs-6'>
             <h4>{{t 'registries.discover.sidebar.active_filters'}}:</h4>
         </div>
-        <div class="col-xs-6">
-            <button data-test-clear-filters class="pull-right btn btn-sm btn-default" {{action 'clearFilters'}}>
+        <div class='col-xs-6'>
+            <button data-test-clear-filters type='button' class='pull-right btn btn-sm btn-default' {{action 'clearFilters'}}>
```
Comment (adlius) [inline lib/registries/addon/components/registries-discover-search/x-sidebar/template.hbs:7]: Split into multiple lines here?

File: lib/registries/addon/components/registries-discover-search/x-sidebar/template.hbs [hunk 17]
```diff
@@ -1,18 +1,18 @@
...
                 {{t 'registries.discover.sidebar.clear_filters'}}
                 {{fa-icon 'times'}}
             </button>
         </div>
     </div>
 
-    {{#each filters as |filter|}}
-        <button class="{{filter.class}}" local-class="Sidebar__ActiveFilter" {{action 'removeFilter' filter.filter}}>
+    {{#each this.filters as |filter|}}
+        <button type='button' class='{{filter.class}}' local-class='Sidebar__ActiveFilter' {{action 'removeFilter' filter.filter}}>
```
Comment (adlius) [inline lib/registries/addon/components/registries-discover-search/x-sidebar/template.hbs:15]: Same here.

File: lib/registries/addon/components/registries-header/template.hbs [hunk 18]
```diff
@@ -14,23 +13,36 @@
...
                     @placeholder={{t 'registries.header.search_placeholder'}}
                 />
                 <span class='input-group-btn'>
-                    {{#if showHelp}}
-                        <button class='btn btn-default' type='button' {{action 'toggleHelp'}}>{{fa-icon 'question' class='text-muted'}}</button>
+                    {{#if this.showHelp}}
+                        <button
+                            class='btn btn-default'
+                            type='button'
+                            {{action 'toggleHelp'}}
```
Comment (adlius) [inline lib/registries/addon/components/registries-header/template.hbs:26]: Could use element modifier here.

File: lib/registries/addon/components/registries-registration-type-facet/template.hbs [hunk 19]
```diff
@@ -1,20 +1,19 @@
-{{! template-lint-disable no-implicit-this }}
-{{#registries-search-facet-container title=title}}
-    <div local-class={{if onlyOSF 'RegistrationType' 'RegistrationType--Disabled'}}>
-        {{#unless onlyOSF}}
+{{#registries-search-facet-container title=this.title}}
```
Comment (adlius) [inline lib/registries/addon/components/registries-registration-type-facet/template.hbs:1]: Angle bracket invocations here?

File: lib/registries/addon/components/registries-search-result/template.hbs [hunk 20]
```diff
@@ -1,30 +1,30 @@
...
-        {{#if result.mainLink}}
+<div class='row'>
+    <div class='col-xs-12'>
+        {{#if @result.mainLink}}
             {{sharing-icons/popover
-                title=result.title
-                description=result.description
-                hyperlink=result.mainLink
-                resultId=result.id
+                title=@result.title
```
Comment (adlius) [inline lib/registries/addon/components/registries-search-result/template.hbs:5]: Angle bracket invocations here?

File: lib/registries/addon/components/registries-search-result/template.hbs [hunk 21]
```diff
@@ -36,61 +36,68 @@
...
+        <div class='text-center'>
+            <button
+                data-test-result-toggle-id={{@result.id}}
+                type='button'
+                local-class='RegistriesSearchResult__Toggle'
+                class='btn btn-link'
+                aria-label={{t 'eosf.components.searchResult.showResult'}}
+                {{action this.toggleExpanded}}
+            >
+                {{fa-icon this.footerIcon size=1 ariaHidden=false}}
```
Comment (adlius) [inline lib/registries/addon/components/registries-search-result/template.hbs:100]: Angle bracket invocations here?

File: lib/registries/addon/components/registries-services-list/template.hbs [hunk 22]
```diff
@@ -1,48 +1,50 @@
...
-                    <a local-class="ServicesList__ExternalLink" href="https://github.com/CenterForOpenScience/ember-osf-registries"
-                        onclick={{action "click" "link" "Index - GitHub Repo" "https://github.com/CenterForOpenScience/ember-osf-registries" target=analytics}}>
-                        {{~t "registries.index.services.bottom.div.linkText1"~}}
+    <div class='row p-v-md'>
+        <div class='col-md-12 text-center'>
+            <p class='lead'>
+                {{t 'registries.index.services.bottom.p1'}}
+                <div class='p-t-sm'>
+                    {{t 'registries.index.services.bottom.div.line1'}}
+                    <a local-class='ServicesList__ExternalLink' href='https://github.com/CenterForOpenScience/ember-osf-registries'
```
Comment (adlius) [inline lib/registries/addon/components/registries-services-list/template.hbs:33]: Multi-line?

File: lib/registries/addon/components/registries-services-list/template.hbs [hunk 23]
```diff
@@ -1,48 +1,50 @@
...
+                        onclick={{action 'click' 'link' 'Index - GitHub Repo' 'https://github.com/CenterForOpenScience/ember-osf-registries' target=this.analytics}}>
+                        {{~t 'registries.index.services.bottom.div.linkText1'~}}
                     </a>
-                    {{t "registries.index.services.bottom.div.line2"}}
-                    <a local-class="ServicesList__ExternalLink" href="https://cos.io/our-products/product-roadmap/"
-                        onclick={{action "click" "link" "Index - public roadmap" "https://cos.io/our-products/product-roadmap/" target=analytics}}>
-                        {{~t "registries.index.services.bottom.div.linkText2"~}}
+                    {{t 'registries.index.services.bottom.div.line2'}}
+                    <a local-class='ServicesList__ExternalLink' href='https://cos.io/our-products/product-roadmap/'
```
Comment (adlius) [inline lib/registries/addon/components/registries-services-list/template.hbs:38]: Multi-line?

File: lib/registries/addon/components/registries-services-list/template.hbs [hunk 24]
```diff
@@ -1,48 +1,50 @@
...
+                        onclick={{action 'click' 'link' 'Index - public roadmap' 'https://cos.io/our-products/product-roadmap/' target=this.analytics}}>
+                        {{~t 'registries.index.services.bottom.div.linkText2'~}}
                     </a>
-                    {{t "registries.index.services.bottom.div.line3"}}
+                    {{t 'registries.index.services.bottom.div.line3'}}
                 </div>
             </p>
-            <a href="mailto:contact@osf.io" class="btn btn-info btn-lg" onclick={{action "click" "link" "Index - Contact" target=analytics}}>{{t "registries.index.services.bottom.contact"}}</a>
+            <a href='mailto:contact@osf.io' class='btn btn-info btn-lg' onclick={{action 'click' 'link' 'Index - Contact' target=this.analytics}}>
```
Comment (adlius) [inline lib/registries/addon/components/registries-services-list/template.hbs:45]: Same here.

File: lib/registries/addon/discover/template.hbs [hunk 25]
```diff
@@ -1,16 +1,21 @@
-{{! template-lint-disable no-implicit-this }}
 {{title (t 'registries.discover.page_title')}}
 
-{{#registries-header showHelp=true value=(mut query) onSearch=(action 'onSearch') searchable=searchable as |header|}}
+{{#registries-header showHelp=true value=(mut this.query) onSearch=(action 'onSearch') searchable=this.searchable as |header|}}
```
Comment (adlius) [inline lib/registries/addon/discover/template.hbs:3]: Angle bracket invocation here?

File: lib/analytics-page/addon/application/styles.scss (hunk 1 above)
Comment (fabmiz) [reply lib/analytics-page/addon/application/styles.scss:61]: Correct.

File: lib/app-components/addon/components/search-paginator/template.hbs (hunk 3 above)
Comment (fabmiz) [reply lib/app-components/addon/components/search-paginator/template.hbs:9]: the `fn` helper can be used.

File: lib/osf-components/addon/components/new-project-modal/template.hbs (hunk 13 above)
Comment (fabmiz) [reply lib/osf-components/addon/components/new-project-modal/template.hbs:54]: I think we want to use `on`  element modifier on HTML elements as opposed to components, because components may have their own way of handling the passed in `click` action.

File: lib/osf-components/addon/components/osf-navbar/x-links/hyper-link/template.hbs (hunk 14 above)
Comment (fabmiz) [reply lib/osf-components/addon/components/osf-navbar/x-links/hyper-link/template.hbs:10]: 🤷🏾‍♂️I don't think it worked, but also there seem to be no test looking for that test selector.

File: translations/en-us.yml (hunk 9 above)
Comment (fabmiz) [reply translations/en-us.yml:1218]: Because we have escapes `\`, YAML syntax requires us to use double quotes else a string `'\n'` will be returned instead of a newline.

File: translations/en-us.yml (hunk 8 above)
Comment (fabmiz) [reply translations/en-us.yml:945]: Quotes are generally not needed unless this is a multi-word string.

File: lib/osf-components/addon/components/sign-up-form/template.hbs (hunk 15 above)
Comment (fabmiz) [reply lib/osf-components/addon/components/sign-up-form/template.hbs:73]: `{{}}` are not needed here because `valuePath` is coerced to string. Basically, `{{}}` are needed for boolean and numbers so they don't get coerced to string unintentionally.

File: lib/app-components/addon/components/project-contributors/list/item/template.hbs (hunk 2 above)
Comment (fabmiz) [reply lib/app-components/addon/components/project-contributors/list/item/template.hbs:114]: I think it's a good idea because of how confusing it's gotten. But as you say, let's defer enabling it to next iteration.

File: lib/osf-components/addon/components/new-project-modal/template.hbs (hunk 13 above)
Comment (aaxelb) [reply lib/osf-components/addon/components/new-project-modal/template.hbs:54]: if the only thing the component is doing with the passed-in `onClick` is its own `{{on 'click' @onClick}}` on the element with `...attributes`, i think it'd be better to use `{{on}}` on the component -- it's nice to treat components the same as normal elements, when we can, and we could get rid of the `onClick` arg to simplify the component.

if there's a more complex component with a passed-in action that might be called for different/multiple reasons, then yeah, that should stay an argument -- but i'd probably want a name that indicates that it's more complex than "on click".

File: lib/osf-components/addon/components/sign-up-form/template.hbs (hunk 15 above)
Comment (adlius) [reply lib/osf-components/addon/components/sign-up-form/template.hbs:73]: I don't like how we have to think what kind of argument we are passing before choosing to use `{{}}` or not. If using `{{}}` on strings doesn't affect anything, might as well make it so that we always use `{{}}` regardless of argument types. We've seen enough confusion on when and when not to use `{{}}` already.

File: translations/en-us.yml (hunk 8 above)
Comment (adlius) [reply translations/en-us.yml:945]: Maybe we enforce the rule of always using quotes?

File: translations/en-us.yml (hunk 9 above)
Comment (adlius) [reply translations/en-us.yml:1218]: Ah... This is annoying. We could consider always using double quotes.

File: lib/registries/addon/components/registries-search-result/template.hbs (hunk 21 above)
Comment (fabmiz) [reply lib/registries/addon/components/registries-search-result/template.hbs:100]: This feels like scope creep. It's really related to the intent of this PR and without removing all the old invocation style, it does not help much. We could consider enabling [`no-curly-component-invocation`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-curly-component-invocation.md) at some point?

Comment (aaxelb) [reply lib/registries/addon/components/registries-search-result/template.hbs:100]: +1 for `no-curly-component-invocation` in another PR

File: translations/en-us.yml (hunk 9 above)
Comment (fabmiz) [reply translations/en-us.yml:1218]: This is beyond scope. Prefer we decide together as a team on this.

File: lib/osf-components/addon/components/sign-up-form/template.hbs (hunk 15 above)
Comment (fabmiz) [reply lib/osf-components/addon/components/sign-up-form/template.hbs:73]: This is beyond scope. Prefer we decide together as a team on this. Will make a note of this.

File: translations/en-us.yml (hunk 8 above)
Comment (fabmiz) [reply translations/en-us.yml:945]: This is beyond scope. Prefer we decide together as a team on this. Will make a note of this.

File: lib/osf-components/addon/components/new-project-modal/template.hbs (hunk 13 above)
Comment (fabmiz) [reply lib/osf-components/addon/components/new-project-modal/template.hbs:54]: I suggest we make this part of the ticket enabling the `no-action` rule. 

Element modifiers are not allowed in components with our version of ember-cli-template-lint. We need to upgrade `ember-cli-template-lint` to `^2.0.2`. `ember-cli-template-lint` is now deprecated and gets dropped in Ember 3.17 in favor `ember-template-lint`.

**Comment by fabmiz** [comment]
Addressed most comments, but in the spirit of keeping the changes close to the intention of the PR (turn on template lint on CI i.e fixing existing rule violations), additional work (enabling new rules) is deferred.
